examples/threading/threading.c

```diff
@@ -2,32 +2,127 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <time.h>
+#include <errno.h>
+#include <syslog.h>
+#include <string.h>
+
+//Defines
+#define NS_TO_MS    (1000000) //Multiply
+#define S_TO_MS     (1000) //Divide
 
 // Optional: use these functions to add debug or error prints to your application
 #define DEBUG_LOG(msg,...)
 //#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
+/**
+ * Implements the functionality stated in threading.h
+ * @param thread_param needs to be parsed to thread_data*
+ * @return the same structure modified depending on the results
+ */
 void* threadfunc(void* thread_param)
 {
+    //Parse the parameter to a thread_data*
+    struct thread_data *parameter = (struct thread_data *)thread_param;
+
+    //Wait (implemented thanks to https://man7.org/linux/man-pages/man2/nanosleep.2.html)
+    struct timespec req = {(parameter->wait_acquire_ms)/S_TO_MS, ((parameter->wait_acquire_ms)%S_TO_MS)*NS_TO_MS};
+    struct timespec rem;
+    int ret = nanosleep(&req, &rem);
+    //Check if an error occurred
+    while(ret == -1)
+    {
+        //Check if the sleep has stopped due to a signal
+        if(errno == EINTR)
+        {
+            syslog(LOG_DEBUG, "Sleep interrupted by a signal, resuming remaining time...");
+            req.tv_sec= 0;
+            req.tv_nsec = rem.tv_nsec;
+            ret = nanosleep(&req, &rem);
+        }
+        //If not, an error occurred and the thread exits
+        else
+        {
+            syslog(LOG_ERR, "Sleep error detected.");
+            parameter->thread_complete_success = false;
+            return (void *)parameter;
+        }
+        
+    }
+
+    //Obtain mutex
+    pthread_mutex_lock(parameter->mutex);
```

Comment:
@jordicrosmompart can you be sure the lock was obtained without an error check here?